### PR TITLE
Add current page header

### PR DESCRIPTION
**Pull request by ceritium**

### CODE REVIEW

**Comment by jgraichen** [comment]
I will have to check if this change works with an internal paginator I'm using where `page` is a complex data structure consisting of a record ID and order direction for tree index based pagination.

It might be necessary to have an extra method returning the "page number value" for the header to make it possible to skip the header.

**Comment by jgraichen** [comment]
It does work. Can you please rebase on master and fix the indention?

File: lib/paginate-responder/base.rb [hunk 1]
```diff
@@ -39,9 +39,10 @@ def update
       link! 'next',  page: next_page  if next_page
       link! 'last',  page: last_page  if last_page
 
-      response.headers["X-Total-Pages"] = total_pages.to_s if total_pages
-      response.headers["X-Total-Count"] = total_count.to_s if total_count
-      response.headers["X-Per-Page"]    = per_page.to_s    if per_page
+      response.headers["X-Total-Pages"]   = total_pages.to_s if total_pages
+      response.headers["X-Total-Count"]   = total_count.to_s if total_count
+      response.headers["X-Per-Page"]      = per_page.to_s    if per_page
+      response.headers["X-Current-Page"]  = page.to_s    if page
```
Comment (jgraichen) [inline lib/paginate-responder/base.rb]: Please fix the two spaces before `=` and the indention in front of the `if`s.

Comment (ceritium) [inline lib/paginate-responder/base.rb]: ok!

**Comment by ceritium** [comment]
@jgraichen I added  this in my project to get 1 instead of 0 if `params[:page]` is `nil`

```ruby
  def page
    (params[:page] || 1).to_i
  end
```

Should we handle this in the gem or delegate it to the host application?

**Comment by jgraichen** [comment]
You sure `params[:page]` is `nil`? This should return `1`:

```ruby
    def page
      @page ||= begin
        val ||= controller.page if controller.respond_to? :page
        val ||= controller.params[:page]
        val ||= first_page # <-- Would call `first_page` if `nil`
        cast_page val
      end
    end
```

**Comment by ceritium** [comment]
Ok, forgot my last comment, it was a problem of my app.

I had:
```ruby
  def page
    params[:page].to_i
  end
```

When `params[:page] => nil` then `page => 0`

The gem is perfect :D

**Comment by jgraichen** [comment]
Release in `1.7.0`.

**Pull request merged**